### ISSUE TITLE
Add proof of fundamental theorem of arithmetic

### ISSUE THREAD

**Issue by rzach** [OpenLogicProject/OpenLogic#189]
computability/recursive-functions/sequences.tex uses the fundamental theorem of arithmetic to code sequences. Should add a proof of this. Since both existence and uniqueness proofs use strong induction, it might fit as an example in methods/induction/strong-induction.tex

**Comment by beastaugh** [comment]
I think this is a good idea, and I've made a start, but the (standard?) proof relies on Euclid's lemma. In the (standard?) modern development of elementary number theory we'd prove Bézout's identity first (this itself requires the development of the idea of the greatest common divisor and coprimality), use that to prove a generalised form of Euclid's lemma, then use that to prove the fundamental theorem of arithmetic.

I think this is a reasonably clean and modular way to do this, and of course each of these results in turn is a good example of strong induction or (equivalently) the least number principle. However, it's not a short one-paragraph example. So we could perhaps consider splitting this out into a separate section on elementary number theory, and add a proof of Sunzi's theorem as well, which is used in the proof of the first incompleteness theorem.

**Comment by rzach** [comment]
Yes! Maybe there should be a methods/number-theory chapter that collects these results and maybe some basics about divisibility and congruences?

**Comment by beastaugh** [comment]
That's more or less what I was thinking, yes.
